confluent_kafka/avro/CachedSchemaRegistryClient.py

```diff
@@ -0,0 +1,288 @@
+                return result
+            except:
+                # bad schema - should not happen
+                self.log.error("Received bad schema from registry.")
+                raise ClientError("Received bad schema from registry.")
+
+    def get_latest_schema(self, subject):
+        """
+        Return the latest 3-tuple of:
+        (the schema id, the parsed avro schema, the schema version)
+        for a particular subject.
+
+        This call always contacts the registry.
+
+        If the subject is not found, (None,None,None) is returned.
+        @:param: subject: subject name
+        @:returns: (schema_id, schema, version)
+        """
+        url = '/'.join([self.url, 'subjects', subject, 'versions', 'latest'])
+
+        result, code = self._send_request(url)
+        if (code != 200):
+            return (None, None, None)
+        schema_id = result['id']
+        version = result['version']
+        if schema_id in self.id_to_schema:
+            schema = self.id_to_schema[schema_id]
+        else:
+            try:
+                schema = Util.parse_schema_from_string(result['schema'])
+            except:
+                # bad schema - should not happen
+                self.log.error("Received bad schema from registry.")
+                raise ClientError("Received bad schema from registry.")
+
+        self._cache_schema(schema, schema_id, subject, version)
+        return (schema_id, schema, version)
+
+    def get_version(self, subject, avro_schema):
+        """
+        Get the version of a schema for a given subject.
+
+        Returns None if not found.
+        @:param: subject: subject name
+        @:param: avro_schema: Avro schema
+        @:returns: version
+        """
+        schemas_to_version = self.subject_to_schema_versions.get(subject, {})
+        version = schemas_to_version.get(avro_schema, None)
+        if version != None:
+            return version
+
+        url = '/'.join([self.url, 'subjects', subject])
+        body = {'schema': json.dumps(avro_schema.to_json())}
+
+        result, code = self._send_request(url, method='POST', body=body)
+        if (code != 200):
+            return None
+        schema_id = result['id']
+        version = result['version']
+        self._cache_schema(avro_schema, schema_id, subject, version)
+        return version
+
+    def test_compatibility(self, subject, avro_schema, version='latest'):
+        """
+        Test the compatibility of a candidate parsed schema for a given subject.
+
+        By default the latest version is checked against.
+        @:param: subject: subject name
+        @:param: avro_schema: Avro schema
+        @:return: True if compatible, False if not compatible
+        """
+        url = '/'.join([self.url, 'compatibility', 'subjects', subject,
+                        'versions', str(version)])
+        avro_schema = body = {'schema': json.dumps(avro_schema.to_json())}
```

Comment:
Why is `avro_schema` being overwritten here? It doesn't look like it gets used again?